internal/fleet/agent_policy/schema.go

```diff
@@ -3,83 +3,112 @@ package agent_policy
-			Optional:    true,
-		},
-		"download_source_id": schema.StringAttribute{
-			Description: "The identifier for the Elastic Agent binary download server.",
-			Optional:    true,
-		},
-		"monitor_logs": schema.BoolAttribute{
-			Description: "Enable collection of agent logs.",
-			Computed:    true,
-			Optional:    true,
-			Default:     booldefault.StaticBool(false),
-		},
-		"monitor_metrics": schema.BoolAttribute{
-			Description: "Enable collection of agent metrics.",
-			Computed:    true,
-			Optional:    true,
-			Default:     booldefault.StaticBool(false),
-		},
-		"skip_destroy": schema.BoolAttribute{
-			Description: "Set to true if you do not wish the agent policy to be deleted at destroy time, and instead just remove the agent policy from the Terraform state.",
-			Optional:    true,
-		},
-		"sys_monitoring": schema.BoolAttribute{
-			Description: "Enable collection of system logs and metrics.",
-			Optional:    true,
-			PlanModifiers: []planmodifier.Bool{
-				boolplanmodifier.RequiresReplace(),
+			"fleet_server_host_id": schema.StringAttribute{
+				Description: "The identifier for the Fleet server host.",
+				Optional:    true,
 			},
-		},
-	}
+			"download_source_id": schema.StringAttribute{
+				Description: "The identifier for the Elastic Agent binary download server.",
+				Optional:    true,
+			},
+			"monitor_logs": schema.BoolAttribute{
+				Description: "Enable collection of agent logs.",
+				Computed:    true,
+				Optional:    true,
+				Default:     booldefault.StaticBool(false),
+			},
+			"monitor_metrics": schema.BoolAttribute{
+				Description: "Enable collection of agent metrics.",
+				Computed:    true,
+				Optional:    true,
+				Default:     booldefault.StaticBool(false),
+			},
+			"skip_destroy": schema.BoolAttribute{
+				Description: "Set to true if you do not wish the agent policy to be deleted at destroy time, and instead just remove the agent policy from the Terraform state.",
+				Optional:    true,
+			},
+			"sys_monitoring": schema.BoolAttribute{
+				Description: "Enable collection of system logs and metrics.",
+				Optional:    true,
+				PlanModifiers: []planmodifier.Bool{
+					boolplanmodifier.RequiresReplace(),
+				},
+			},
+			"global_data_tags": schema.MapNestedAttribute{
+				Description: "User-defined data tags to apply to all inputs. Values can be strings (string_value) or numbers (number_value) but not both. Example -- key1 = {string_value = value1}, key2 = {number_value = 42}",
+				NestedObject: schema.NestedAttributeObject{
+					Attributes: map[string]schema.Attribute{
+						"string_value": schema.StringAttribute{
+							Description: "String value for the field. If this is set, number_value must not be defined.",
+							Optional:    true,
+						},
+						"number_value": schema.Float32Attribute{
+							Description: "Number value for the field. If this is set, string_value must not be defined.",
+							Optional:    true,
+						},
```

Comment:
```suggestion
						"string_value": schema.StringAttribute{
							Description: "String value for the field. If this is set, number_value must not be defined.",
							Optional:    true,
							Validators: []validator.String{
								stringvalidator.ConflictsWith(path.MatchRelative().AtName("number_value")),
							},
						},
						"number_value": schema.Float32Attribute{
							Description: "Number value for the field. If this is set, string_value must not be defined.",
							Optional:    true,
							Validators: []validator.Float32{
								float32validator.ConflictsWith(path.MatchRelative().AtName("number_value")),
							},
						},
```

Are you able to add an acceptance test verifying that this validation is enforced as well?

Reply:
All set, with a new test to test a bad global_data_tag setup